include/server/api/table_parameter_grammar.hpp

```diff
@@ -66,8 +66,12 @@ struct TableParametersGrammar : public BaseParametersGrammar<Iterator, Signature
         annotations.add("duration", AnnotationsType::Duration)("distance",
                                                                AnnotationsType::Distance);
 
+        const auto reset_annotations = [](engine::api::TableParameters &table_parameters) {
```

Comment:
@ghoshkaj minor thing: you can avoid using explicit lambdas by using a rule for an annotations list as
```
        annotations.add
            ("duration", AnnotationsType::Duration)
            ("distance", AnnotationsType::Distance)
            ;

        annotations_list = annotations[qi::_val = qi::_val | qi::_1]  % ','
            ;

        base_rule = BaseGrammar::base_rule(qi::_r1)
            | (qi::lit("annotations=") > annotations_list
               [ph::bind(&engine::api::TableParameters::annotations, qi::_r1) = qi::_1])
            ;
```

and below in the members section
```
    qi::rule<Iterator, engine::api::TableParameters::AnnotationsType()> annotations_list;
```

It is also possible to use `qi::_val |=  qi::_1` instead of `qi::_val = qi::_val | qi::_1` in the semantic action for `annotations_list`, but it requires a fix at
https://github.com/Project-OSRM/osrm-backend/blob/ce5cfc4b6997eab0df34ba93f107dd066de7bd11/include/engine/api/table_parameters.hpp#L138
as
```
inline TableParameters::AnnotationsType& operator|=(TableParameters::AnnotationsType& lhs,
```